ixmp4/data/db/optimization/table/filter.py

```diff
@@ -0,0 +1,11 @@
+from ixmp4.data.db import filters as base
+from ixmp4.db import filters
+
+
+class OptimizationTableFilter(
+    base.OptimizationTableFilter, metaclass=filters.FilterMeta
+):
+    table: base.OptimizationTableFilter | None
```

Comment:
The table filter has an item named table which is the same filter again?

Reply:
This was adapted from Scalar, which came from Indexset and at some point, it was probably necessary to have this in. I have to admit that I'm not entirely sure why the current way of writing these filters is better, but of course I'll update/clean up the optimization filters.

Reply:
Not sure im completely follwing right now, in this specific case wouldn't you just remove the table: ... annotation?

Reply:
Yes, I did, but there were other changes introduced by your recent #56. See e.g. https://github.com/iiasa/ixmp4/blob/bcbdff3522f7d61be4d0d21060336238199bd0ed/ixmp4/data/db/optimization/indexset/filter.py#L8-L19 and compare to [how it was before](https://github.com/iiasa/ixmp4/blob/7798462920e7f06167163d3380f092c310581ae2/ixmp4/data/db/optimization/indexset/filter.py). I've just made the same changes now for all Optimization*Filters assuming that they represent a performance improvement.